tests/foreman/ui_airgun/test_bookmark.py

```diff
@@ -0,0 +1,130 @@
+import random
+
+from airgun.session import Session
+from fauxfactory import gen_string
+from nailgun import entities
+
+from robottelo.constants import BOOKMARK_ENTITIES
+from robottelo.decorators import bz_bug_is_open, fixture, rm_bug_is_open, tier2
+
+
+@fixture(scope='module')
+def module_org():
+    return entities.Organization().create()
+
+
+@fixture(scope='module')
+def custom_user(module_org):
+    """Custom user with viewer role for tests validating 'public' bookmark
+    option.
+    """
+    viewer_role = entities.Role().search(query={'search': 'name="Viewer"'})[0]
+    bm_role = entities.Role().create()
+    entities.Filter(
+        permission=entities.Permission(
+            resource_type='Bookmark').search(
+            filters={'name': 'view_bookmarks'}),
+        role=bm_role,
+    ).create()
+    custom_password = gen_string('alphanumeric')
+    custom_user = entities.User(
+        default_organization=module_org,
+        organization=[module_org],
+        role=[viewer_role, bm_role],
+        password=custom_password,
+    ).create()
+    custom_user.password = custom_password
+    return custom_user
+
+
+@fixture(scope='module')
+def ui_entities(module_org):
+    """Collects the list of all applicable UI entities for testing and does all
+    required preconditions.
+    """
+    ui_entities = []
+    for entity in BOOKMARK_ENTITIES:
+        # Skip the entities, which can't be tested ATM (not implemented in
+        # airgun or have open BZs)
+        skip = entity.get('skip_for_ui')
+        if isinstance(skip, tuple):
+            if (skip[0] == 'bugzilla' and bz_bug_is_open(skip[1])
+                    or skip[0] == 'redmine' and rm_bug_is_open(skip[1])):
+                skip = True
+        if skip is True:
+            continue
+        ui_entities.append(entity)
+        # Some pages require at least 1 existing entity for search bar to
+        # appear. Creating 1 entity for such pages
+        if entity.get('setup'):
```

Comment:
perhaps like this , we do not have to worry about creating new local variables
```python
entity_name, entity_setup =   entity['name'], entity.get('setup')
if entity_setup:
      # entities with 1 organization
      if entity_name in ('Host',):
          entity_setup(organization=module_org).create()
    ...
```

Reply:
Could you explain your point? And i can't see any benefits in creating extra variables when they aren't needed (don't save us time or operations, aren't more readable etc).

Reply:
why we have to retrieve/access that vars form dict all the time (think retrieve them ones)  are we afraid to create 2 new local vars for that?

Reply:
Variable is just a pointer to that dict key, so we will access it anyway. What's the real value of accessing the key through extra pointer? :)

Reply:
the variables here are not a pointer to the dict key, but a pointer to the value of the dict key.